pkg/envoy/resources.go

```diff
@@ -118,7 +118,7 @@ func (cache *NPHDSCache) handleIPDelete(npHost *envoyAPI.NetworkPolicyHosts, pee
 		}
 	}
 	if targetIndex < 0 {
-		scopedLog.Warning("Can't find IP in NPHDS cache")
+		scopedLog.Fatal("Can't find IP in NPHDS cache")
```

Comment:
I might miss some context but how does killing the agent fixes the bug?

Reply:
This is a safety measure, complementary to the real fix in the other file.

When this condition is hit, it means that there's a bug in the Cilium agent's code that syncs the ipcache with the NPHDS cache. Right now when such a bug manifests, Envoy is stuck, resulting in loss of traffic and the impossibility to configure new L7 policies. And currently the only way to work around the issue is to manually kill the Cilium agent. This change does it automatically.